### PR TITLE
Fixing transfer nft logic through the Node

### PR DESCRIPTION
**Pull request by aaitor**
## Description

- fix: sending did for transfer nft through Node
- chore: bumping to v1.3.13

## Is this PR related with an open issue?

Related to Issue #

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] Follows the code style of this project.
- [ ] Tests Cover Changes
- [ ] Documentation

#### Funny gif

Put a link of a funny gif inside the parenthesis-->

### CODE REVIEW

File: src/nevermined/api/RegistryBaseApi.ts [hunk 1]
```diff
@@ -586,6 +586,17 @@ export abstract class RegistryBaseApi extends Instantiable {
         throw new AssetError(`Error creating ${serviceType} agreement`)
       }
 
+      // Checking the agreementId was created on-chain with the correct DID associated to it
+      const agreementData = await this.nevermined.keeper.agreementStoreManager.getAgreement(
```
Comment (r-marques) [inline src/nevermined/api/RegistryBaseApi.ts:590]: `agreementStoreManager.getAgreement` is using events. We should call https://github.com/nevermined-io/contracts/blob/main/contracts/templates/BaseEscrowTemplate.sol#LL173 to check if an agreement exists without using events

File: src/services/node/NeverminedNode.ts [hunk 2]
```diff
@@ -380,10 +380,16 @@ export class NeverminedNode extends Instantiable {
     ercType: ERCType = 1155,
   ): Promise<boolean> {
     try {
+      // We get the DID on-chain to make sure the DID and agreement id exist
+      const agreementData = await this.nevermined.keeper.agreementStoreManager.getAgreement(
```
Comment (r-marques) [inline src/services/node/NeverminedNode.ts:384]: same as above

File: src/services/node/NeverminedNode.ts [hunk 3]
```diff
@@ -380,10 +380,16 @@ export class NeverminedNode extends Instantiable {
...
+        agreementId,
+      )
+      console.log(`Agreement DATA: ${JSON.stringify(agreementData)}`)
```
Comment (r-marques) [inline src/services/node/NeverminedNode.ts:387]: cleanup

**Review by r-marques — APPROVED** [review]

File: src/nevermined/api/RegistryBaseApi.ts [hunk 4]
```diff
@@ -586,6 +592,33 @@ export abstract class RegistryBaseApi extends Instantiable {
...
+          )
+      } else if (serviceType.startsWith('nft')) {
+        agreementData = await this.nevermined.keeper.templates.nftSalesTemplate.getAgreementData(
+          agreementId,
+        )
+      } else {
+        agreementData = await this.nevermined.keeper.templates.accessTemplate.getAgreementData(
+          agreementId,
+        )
+      }
```
[the remark this answers is not in the document]
Comment (aaitor) [reply src/nevermined/api/RegistryBaseApi.ts]: refactored

**Pull request merged**